packages/deployments/contracts/contracts/core/connext/helpers/Executor.sol

```diff
@@ -227,6 +227,19 @@ contract Executor is IExecutor {
       _sendToRecovery(isNative, hasValue, _args.assetId, payable(_args.to), payable(_args.recovery), _args.amount);
     }
 
+    // If there is unclaimed tokens, send them to the recovery address
+    if (hasValue) {
+      uint256 unclaimed = _args.amount - (beforeBalance - _assetBalance(_args.assetId));
+      if (unclaimed > 0) {
+        _sendToRecovery(isNative, hasValue, _args.assetId, payable(_args.to), payable(_args.recovery), unclaimed);
+      }
```

Comment:
This will decrease allowance as well. so i think we don't need `safeApprove` anymore